test/runtests.jl

```diff
@@ -1612,6 +1625,77 @@ end
             close(conn)
             @test !isopen(conn)
         end
+
+        @testset "getindex(::Column) performance" begin
+            @testset "$value" for (value, result) in test_data_1()
+
+                # Establish connection and construct temporary table.
+                conn = LibPQ.Connection("dbname=postgres user=$DATABASE_USER")
+                result = execute(conn, """
+                    CREATE TEMPORARY TABLE libpqjl_test (
+                        id bigint,
+                        my_string_column varchar
+                    );
+                """)
+
+                # Put some data in the table.
+                execute(
+                    conn, "INSERT INTO libpqjl_test (my_string_column) VALUES ($value)"
+                )
+
+                # Get the column.
+                result = execute(
+                    conn, "SELECT my_string_column FROM libpqjl_test"
+                )
+                col = columntable(result).my_string_column
+
+                # Ensure that getting an element from the column produces exactly 1 alloc.
+                foo(col) = col[1]
+                count_allocs(foo, col)
+                @test count_allocs(foo, col) == 2
+
+                close(result)
+                close(conn)
+            end
+            @testset "$value" for (value, pg_type_string) in [
+                (5.0, "float8"),
+                (5f0, "float4"),
+                (Int64(3), "int8"),
+                (Int32(3), "int4"),
+                (Int16(3), "int2"),
+            ]
+                # Establish connection and construct temporary table.
+                conn = LibPQ.Connection("dbname=postgres user=$DATABASE_USER")
+                result = execute(conn, "
+                    CREATE TEMPORARY TABLE libpqjl_test (
+                        id bigint,
+                        my_column $pg_type_string
+                    );
+                ")
+
+                # Put some data in the table.
+                execute(
+                    conn, "INSERT INTO libpqjl_test (my_column) VALUES ($value)"
+                )
+
+                # Get the column.
+                result = execute(
+                    conn, "SELECT my_column FROM libpqjl_test"; binary_format=true,
```

Comment:
You should repeat this `@testset` with `binary_format=false` and measure that as well, and add more complicated types like maybe `Time`.

I think this could be expanded more but I don't think it's the highest priority.

Reply:
I've not managed to add a time-related quantity to the test -- if you're content with the new state, please resolve this thread :)

Reply:
Have added a string test and removed the previous tests

Reply:
Looks like you commented out a large swath of code so it's a bit hard to read

Reply:
Oops, apologies. Is now uncommented.